src/MyFrame.cpp

```diff
@@ -2025,6 +2038,35 @@ MyFrame::OnRevealLetter(wxCommandEvent & WXUNUSED(evt))
 }
 
 void
+MyFrame::OnConvertToNormal(wxCommandEvent & WXUNUSED(evt))
+{
+    wxASSERT(m_puz.IsDiagramless());
+
+    if (! XWordPrompt(this, _T("This will convert this diagramless puzzle to a normal puzzle, ")
+                            _T("revealing all black squares and overwriting any letters entered ")
+                            _T("where black squares should be.  Continue?")))
+        return;
+
+    m_puz.ConvertDiagramlessToNormal();
+
+    // Turn off the diagramless menu items
+    EnableDiagramless(false);
+
+    // Ensure focus moves to a different word as it may be on a black square. We move twice to
+    // ensure that the old word is cleared, as it may not have the correct boundaries.
+    m_XGridCtrl->SetFocusedSquare(m_XGridCtrl->LastWhite(), puz::ACROSS);

```

Comment:
XGridCtrl::SetFocusedSquare tries to re-use the currently focused word if you don't supply it a new word, and it doesn't check to make sure it's a valid word (since changing from diagramless to normal isn't an expected behavior).  The easiest fix is to supply the word to SetFocusedSquare:

```c++
puz::Square * square = m_XGridCtrl->GetFocusedSquare();
if (square == NULL || square->IsBlack())
    square = m_XGridCtrl->FirstWhite();
m_XGridCtrl->SetFocusedSquare(square, m_puz.FindWord(square));
```